build_tools/buildkite/pipelines/postsubmit.yml

```diff
@@ -0,0 +1,38 @@
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+agents:
+  queue: "orchestration"
+  security: "submitted"
+
+steps:
+  - label: ":hiking_boot: Bootstrapping postsubmit pipeline"
+    env:
+      PIPELINE_BOOTSTRAPPED: ${PIPELINE_BOOTSTRAPPED:-false}
+    commands: |
+      ./build_tools/buildkite/scripts/bootstrap_pipeline.sh \
+          build_tools/buildkite/pipelines/postsubmit.yml
+
+  - wait
+
+  - label: "Checking for later builds of this pipeline"
+    commands: |
+      export BUILDKITE_ACCESS_TOKEN="$(gcloud secrets versions access latest \
+          --secret=iree-buildkite-privileged)"
+      # Note that dollar sign has to be escaped to avoid Buildkite trying to do
+      # a substitution.
+      ./build_tools/buildkite/scripts/check_for_later_build.py || ret=\$?
+      if (( ret == 2 )); then
+        buildkite-agent pipeline upload \
+            ./build_tools/buildkite/pipelines/update_pipelines.yml
```

Comment:
Could note that return code 2 is for a later build existing and explain why a later build existing should trigger a pipeline update. I was going to suggest putting return codes in a `constants.py`/`return_codes.py` file but using from languages like yaml isn't so easy :P

---

Actually, I'm trying to imagine the different timing/race conditions that can occur here and I don't understand this fully.

1) If I'm working on a PR (not touching the pipelines), that PR will run presubmits from the pipelines at the tip of https://github.com/google/iree/tree/main? So if the pipelines change while I'm still working on that PR then retriggering them may run different builds (say, if someone adds a new GPU pipeline)?
2) If two PRs update the pipelines themselves, is it possible for the first commit that lands to run second, undoing the uploaded changes from the second PR?
3) If a PR updating the pipelines is submitted with no other recent commits, wouldn't this "check for a later build" return 0, so the pipelines aren't updated?

Reply:
Well I'm glad you flagged this because a later build existing should mean we *don't* trigger an update :facepalm: